packages/sign_in_with_apple/android/src/main/kotlin/com/aboutyou/dart_packages/sign_in_with_apple/SignInWithApplePlugin.kt

```diff
@@ -53,62 +56,91 @@ public class SignInWithApplePlugin: FlutterPlugin, MethodCallHandler, ActivityAw
     @JvmStatic
     fun registerWith(registrar: Registrar) {
       val channel = MethodChannel(registrar.messenger(), "com.aboutyou.dart_packages.sign_in_with_apple")
-      channel.setMethodCallHandler(SignInWithApplePlugin())
+      val plugin = SignInWithApplePlugin()
+      channel.setMethodCallHandler(plugin)
+
+      registrar.addActivityResultListener(plugin)
+      registrar.activity()?.let(plugin::registerActivity)
+      registerLifecycle(application(registrar), plugin)
+    }
+
+    private fun registerLifecycle(app: Application, plugin: SignInWithApplePlugin) {
+      app.registerActivityLifecycleCallbacks(object : Application.ActivityLifecycleCallbacks {
+        override fun onActivityPaused(activity: Activity) = Unit
+        override fun onActivityResumed(activity: Activity) = Unit
+        override fun onActivityDestroyed(activity: Activity?) = plugin.unregisterActivity()
+        override fun onActivitySaveInstanceState(activity: Activity?, outState: Bundle?) = Unit
+        override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) = plugin.registerActivity(activity)
+        override fun onActivityStarted(activity: Activity) = Unit
+        override fun onActivityStopped(activity: Activity) = Unit
+      })
     }
+
+    private fun application(registrar: Registrar) = registrar.context().applicationContext as Application
```

Comment:
If I'm not mistaken, these `application` and `registerLifecycle`  are not used anywhere else. What do you think about inlining them to keep the deprecated support in a single function?

Reply:
I'm a fun of small functions. But understand that you want to at one time just wype everything at once.
Will change.